metrics/release_test.go

```diff
@@ -0,0 +1,221 @@
+		metrics.Registry.Unregister(ReleaseAttemptRunningSeconds)
+		metrics.Registry.Unregister(ReleaseAttemptConcurrentTotal)
+		metrics.Registry.Unregister(ReleaseAttemptDurationSeconds)
+	})
+
+	var (
+		AttemptRunningSecondsHeader = inputHeader{
+			Name: "release_attempt_running_seconds",
+			Help: "Release durations from the moment the release resource was created til the release is marked as running",
+		}
+		AttemptDurationSecondsHeader = inputHeader{
+			Name: "release_attempt_duration_seconds",
+			Help: "Release durations from the moment the release PipelineRun was created til the release is marked as finished",
+		}
+		AttemptTotalHeader = inputHeader{
+			Name: "release_attempt_total",
+			Help: "Total number of releases processed by the operator",
+		}
+	)
+
+	const (
+		validReleaseReason   = "valid_release_reason"
+		invalidReleaseReason = "invalid_release_reason"
+		strategy             = "nostrategy"
+	)
+
+	var defaultNamespace = kcp.NamespaceReference{
+		Namespace: "default",
+		Workspace: "default",
+	}
+
+	Context("When RegisterNewRelease is called", func() {
+
+		// As we need to share metrics within the Context, we need to use "per Context" `Before`|`After`All()
+		BeforeAll(func() {
+			// Overriding the metric definitions to simplify the tests
+			//
+			// `Help` can't be overriden due to `https://github.com/prometheus/client_golang/blob/83d56b1144a0c2eb10d399e7abbae3333bebc463/prometheus/registry.go#L314`
+			ReleaseAttemptRunningSeconds = prometheus.NewHistogram(
+				prometheus.HistogramOpts{
+					Name:    "release_attempt_running_seconds",
+					Help:    "Release durations from the moment the release resource was created til the release is marked as running",
+					Buckets: []float64{1, 5, 10, 30},
+				},
+			)
+			ReleaseAttemptConcurrentTotal = prometheus.NewGauge(
+				prometheus.GaugeOpts{
+					Name: "release_attempt_concurrent_requests",
+					Help: "Total number of concurrent release attempts",
+				},
+			)
+			metrics.Registry.MustRegister(ReleaseAttemptRunningSeconds,
+				ReleaseAttemptConcurrentTotal)
+
+		})
+
+		AfterAll(func() {
+			metrics.Registry.Unregister(ReleaseAttemptRunningSeconds)
+			metrics.Registry.Unregister(ReleaseAttemptConcurrentTotal)
+
+		})
+
+		// Buckets for operation duration less or equal to 1, 5, 10 and 30 seconds
+		buckets := []float64{1, 5, 10, 30}
+		elapsedSeconds := 0.0
+
+		It(`increments the 'release_attempt_concurrent_total'.`, func() {
+			creationTime := metav1.Time{}
+			for _, bucket := range buckets {
+				startTime := metav1.NewTime(creationTime.Add(time.Second * time.Duration(bucket)))
+				elapsedSeconds += bucket
+				RegisterNewRelease(creationTime, &startTime)
+			}
+			Expect(testutil.ToFloat64(ReleaseAttemptConcurrentTotal)).To(Equal(float64(len(buckets))))
+		})
+
+		It(`registers a new observation for 'release_attempt_running_seconds' with the elapsed time from the moment
```

Comment:
I think that as you didn't use a global BeforeEach, you fell in the same trap. Imagine that I want to run this tests in an isolated fashion. I can't. It's not even calling `RegisterNewRelease`. You are relying in the previous test.